article.md

```diff
@@ -72,19 +122,24 @@ if __name__ == "__main__":
 
 [^ndbp_aiohttp]: Bibliothèque qui propose des fonctions d'entrées/sorties sur le protocole http.
 
+Dans notre exemple la coroutine est suspendu explicitement à plusieurs endroits (partout où est noté l'expression `yield from`) et la boucle événementielle (définit à la ligne 27) pourrait ainsi en profiter pour faire d'autres opérations. Les générateurs et *asyncio* permettent donc de découper facilement les opérations d'une fonction entre les entrées/sorties mais on peut remarquer quelques problèmes rendant le code moins clair :
+
+ - Il y a détournement du rôle d'origine de l'instruction `yield from`. Sans le décorateur la fonction pourrait être facilement confondue avec un générateur classique.
+ - Tandis que l'exemple d'origine utilisait `with` pour assurer la fermeture du fichier même en cas d'exception, ici nous somme obligé de protéger "manuellement" le code, `with` n'étant pas prévu pour être appelé comme un générateur.
+ - De la même façon il est nécessaire de boucler "manuellement" sur le contenu, l'expression `for` n'étant pas prévu pour ce genre de fonctionnement.
```

Comment:
le `async for` c'est différent. Ça revient à faire un `await
obj.__aiter__()` pour récupérer les différentes valeurs.

Christophe.

2015-08-30 13:55 GMT+02:00 Vincent <notifications@github.com>:

> In article.md
> <https://github.com/Vayel/Python-3.5/pull/17#discussion_r38270961>:
>
> > @@ -72,19 +122,24 @@ if __name__ == "__main__":
> >
> >  [^ndbp_aiohttp]: Bibliothèque qui propose des fonctions d'entrées/sorties sur le protocole http.
> >
> > +Dans notre exemple la coroutine est suspendu explicitement à plusieurs endroits (partout où est noté l'expression `yield from`) et la boucle événementielle (définit à la ligne 27) pourrait ainsi en profiter pour faire d'autres opérations. Les générateurs et *asyncio* permettent donc de découper facilement les opérations d'une fonction entre les entrées/sorties mais on peut remarquer quelques problèmes rendant le code moins clair :
> > +
> > + - Il y a détournement du rôle d'origine de l'instruction `yield from`. Sans le décorateur la fonction pourrait être facilement confondue avec un générateur classique.
> > + - Tandis que l'exemple d'origine utilisait `with` pour assurer la fermeture du fichier même en cas d'exception, ici nous somme obligé de protéger "manuellement" le code, `with` n'étant pas prévu pour être appelé comme un générateur.
> > + - De la même façon il est nécessaire de boucler "manuellement" sur le contenu, l'expression `for` n'étant pas prévu pour ce genre de fonctionnement.
>
> Je ne comprends pas trop cela. On peut bien mettre un yield from dans une
> boucle for, non ? Du moins, ici
> <http://sametmax.com/comment-utiliser-yield-et-les-generateurs-en-python/>,
> on peut mettre un yield.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Vayel/Python-3.5/pull/17/files#r38270961>.
>